Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+                .unregisterReceiver(eventReceiver);
+
+		mBDLocClient.stop();                       // close baidu location service
+		getBaiduMap().setMyLocationEnabled(false); // disable location layer
+    }
+
+    @Override
+    public void clearFlightPath() {
+        if (mFlightPath != null) {
+            List<LatLng> oldFlightPath = mFlightPath.getPoints();
+            oldFlightPath.clear();
+            mFlightPath.setPoints(oldFlightPath);
+        }
+    }
+
+    @Override
+    public void downloadMapTiles(MapDownloader mapDownloader, VisibleMapArea mapRegion, int minimumZ, int maximumZ) {
+    }
+
+    @Override
+    public LatLong getMapCenter() {
+        return LatLngToCoord(getBaiduMap().getMapStatus().target);
+    }
+
+    @Override
+    public float getMapZoomLevel() {
+        return getBaiduMap().getMapStatus().zoom;
+    }
+
+    @Override
+    public float getMaxZoomLevel() {
+        return getBaiduMap().getMaxZoomLevel();
+    }
+
+    @Override
+    public float getMinZoomLevel() { return getBaiduMap().getMinZoomLevel(); }
+
+    @Override
+    public void selectAutoPanMode(AutoPanMode target) {
+        final AutoPanMode currentMode = mPanMode.get();
+        if (currentMode == target) return;
+
+        setAutoPanMode(currentMode, target);
+    }
+
+    private void setAutoPanMode(AutoPanMode current, AutoPanMode update) {
+        if (mPanMode.compareAndSet(current, update)) {
+            switch (current) {
+                case DRONE:
+                    LocalBroadcastManager.getInstance(getActivity().getApplicationContext())
+                            .unregisterReceiver(eventReceiver);
+                    break;
+
+                case DISABLED:
+                default:
+                    break;
+            }
+
+            switch (update) {
+                case DRONE:
+                    LocalBroadcastManager.getInstance(getActivity().getApplicationContext()).registerReceiver
+                            (eventReceiver, mEventFilter);
+                    break;
+
+                case DISABLED:
+                default:
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public DPMapProvider getProvider() {
+        return DPMapProvider.BAIDU_MAP;
+    }
+
+    @Override
+    public void addFlightPathPoint(LatLong coord) {
+        final LatLng position = CoordToLatLang(coord);
+
+        if (maxFlightPathSize > 0) {
+            if (mFlightPath == null) {
+                PolylineOptions flightPathOptions = new PolylineOptions();
+                flightPathOptions.color(FLIGHT_PATH_DEFAULT_COLOR)
+                        .width(FLIGHT_PATH_DEFAULT_WIDTH).zIndex(1);
+                mFlightPath = (Polyline)getBaiduMap().addOverlay(flightPathOptions);
+            }
+
+            List<LatLng> oldFlightPath = mFlightPath.getPoints();
+            if (oldFlightPath.size() > maxFlightPathSize) {
```

Comment:
I'll check it.

BTW, the implementation of addFlightPathPoint() is almost the same as Google Map. 
Doesn't Google map need the while loop?